File: 2021/tests/exercises/test_exercise_11.py
from adventofcode.exercises import exercise_11
from adventofcode import utils

TEST_DATA = """5483143223
2745854711
5264556173
6141336146
6357385478
4167524645
2176841721
6882881134
4846848554
5283751526"""


def test_example_on_exercise_a():
    assert exercise_11.run_a(TEST_DATA) == 1656


def test_returns_correct_answer_on_exercise_a():
    input_data = utils.load_data(exercise_11.EXERCISE)
    assert exercise_11.run_a(input_data) == 1717


def test_example_on_exercise_b():
    assert exercise_11.run_b(TEST_DATA) == 195


def test_returns_correct_answer_on_exercise_b():
    input_data = utils.load_data(exercise_11.EXERCISE)
    assert exercise_11.run_b(input_data) == 476

File: 2024/adventofcode/tests/test_exercise_11.rs
use adventofcode::exercises::exercise_11;
use adventofcode::exercises::utils;

const TEST_DATA: &str = "125 17";

#[test]
fn test_a_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_11::run_a(TEST_DATA), 55312);
}

#[test]
fn test_a_returns_correct_answer_on_exercise_a() {
    let input_data = utils::load_data(exercise_11::EXERCISE);
    assert_eq!(exercise_11::run_a(&input_data), 185894);
}

#[test]
fn test_b_returns_correct_answer_on_exercise_b() {
    let input_data = utils::load_data(exercise_11::EXERCISE);
    assert_eq!(exercise_11::run_b(&input_data), 221632504974231);
}
